ubuntu/xenial/Dockerfile

```diff
@@ -40,5 +40,16 @@ RUN apt-get update && apt-get install -y --force-yes \
     alien \
     dh-systemd
 
+# Install gcc-8
+RUN apt-get update -y && \
+    apt-get upgrade -y &&  \
+    apt-get dist-upgrade -y &&  \
+    apt-get install build-essential software-properties-common zlib1g-dev -y &&  \
+    add-apt-repository ppa:ubuntu-toolchain-r/test -y &&  \
+    apt-get update -y &&  \
+    apt-get install gcc-8 g++-8 -y &&  \
+    update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-8 60 --slave /usr/bin/g++ g++ /usr/bin/g++-8 &&  \
+    update-alternatives --config gcc
```

Comment:
I have a couple of questions.

The repository contains gcc-9 for xenial (but not 10 or 11). What is the reason behind choosing gcc-8?

Some packages are already installed beforehand: at least `build-essential` and `software-properties-common`. Is it necessary to repeat the installation command or it is by a mistake?

Why `zlib` is needed?

Why we need a second `apt-get update`?

Why we need `upgrade` and `dist-upgrade` after `update`?

Shouldn't we copy gcc's package to our backports repository and enable it instead of a testing repository from Ubuntu folks? I guess that they have no commitment to keep all the packages forever.

More on it: are there packages in the added repository that replaces the standard ones? IOW, if I do `apt-get update` inside the container, only `gcc-8` and `g++-8` come from this repository?

`update-alternatives --config` is an interactive command (according to the man page). Is it non-interactive in presence of `DEBIAN_FRONTEND=noninteractive`? if so, what exactly it is doing?

I hope that the resulting patch will answer all the questions on its own: in the comments in the file or in the commit message.

Reply:
I removed all redundant updates, upgrades etc.

Reply:
>>update-alternatives --config is an interactive command (according to the man page). Is it non-interactive in presence of DEBIAN_FRONTEND=noninteractive? if so, what exactly it is doing?

This command works interactive only without parameters. In our case, all parameters set correct.